support/spray/src/test/scala/sjsonnew/support/spray/CalendarFormatsSpec.scala

```diff
@@ -20,16 +20,59 @@ package support.spray
 import spray.json.{ JsValue, JsNumber, JsString, JsNull, JsTrue, JsFalse, JsObject }
 import org.specs2.mutable._
 import java.util.{ Calendar, TimeZone }
+import java.time._
 
 class CalendarFormatsSpec extends Specification with BasicJsonProtocol {
   "The dateStringIso" should {
+    // JDK 8 / Joda dates
+    val zdt = ZonedDateTime.of(1999, 1, 1, 0, 0, 0, 0, ZoneId.of("UTC"))
+    val millis = zdt.plusNanos(1000 * 1000)
+    // zoneless instant in time
+    val inst = millis.toInstant
+    val ld = LocalDate.of(1999, 1, 1)
+
+    // legacy dates
     val seconds = Calendar.getInstance(TimeZone.getTimeZone("UTC"))
     seconds.clear
     seconds.set(1999, 1, 1, 0, 0, 0)
+    
     val milliseconds = Calendar.getInstance(TimeZone.getTimeZone("UTC"))
     milliseconds.clear
     milliseconds.set(1999, 1, 1, 0, 0, 0)
     milliseconds.set(Calendar.MILLISECOND, 1)
+
+    "convert a ZonedDateTime to JsString" in {
+      Converter.toJsonUnsafe(zdt) mustEqual JsString("1999-01-01T00:00:00Z")
+    }
+
+    "convert the JsString back to the ZonedDateTime" in {
+      Converter.fromJsonUnsafe[ZonedDateTime](JsString("1999-01-01T00:00:00Z")).toInstant mustEqual zdt.toInstant
+    }
+
+    "convert a ZonedDateTime with milliseconds to JsString" in {
+      Converter.toJsonUnsafe(millis) mustEqual JsString("1999-01-01T00:00:00.001Z")
+    }
+
+    "convert the JsString back to the ZonedDateTime with milliseconds" in {
+      Converter.fromJsonUnsafe[ZonedDateTime](JsString("1999-01-01T00:00:00.001Z")).toInstant mustEqual millis.toInstant
+    }
+
+    "convert an Instant to JsString" in {
+      Converter.toJsonUnsafe(inst) mustEqual JsString("1999-01-01T00:00:00.001Z")
+    }
+
+    "convert the JsString back to the ZonedDateTime with milliseconds" in {
+      Converter.fromJsonUnsafe[Instant](JsString("1999-01-01T00:00:00.001Z")) mustEqual inst
+    }
+
+    "convert a LocalDate to JsString" in {
+      Converter.toJsonUnsafe(ld) mustEqual JsString("1999-01-01")
+    }
+    
+    "convert the JsString with no time back to the LocalDate" in {
+      Converter.fromJsonUnsafe[LocalDate](JsString("1999-01-01")) mustEqual ld
+    }
+
```

Comment:
That's my oversight. Added it in https://github.com/eed3si9n/sjson-new/pull/54/commits/0b2912768e49adf53ceb573b8df48594e36dd727